src/services/gerente.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@nestjs/common';
+import { Gerente } from 'src/modules/gerente.module';
+import { Cliente } from 'src/modules/cliente.module';
+import { Conta } from 'src/modules/conta.module';
+
+@Injectable()
+export class GerentesService {
+    private gerentes: Gerente[] = [];
+
+    criarGerente(nome: string, id: number): Gerente {
+        const gerente = new Gerente(nome, id);
+        this.gerentes.push(gerente);
+        return gerente;
+    }
+
+    buscarGerentePorId(id: number): Gerente {
+        return this.gerentes.find(gerente => gerente.id === id);
+    }
+
+    adicionarCliente(gerenteId: number, cliente: Cliente): void {
+        const gerente = this.buscarGerentePorId(gerenteId);
+        gerente.adicionarCliente(cliente);
+    }
+
+    removerCliente(gerenteId: number, clienteId: number): void {
+        const gerente = this.buscarGerentePorId(gerenteId);
+        gerente.removerCliente(clienteId);
+    }
+
+    abrirConta(gerenteId: number, tipo: 'corrente' | 'poupanca', clienteId: number): Conta {
+        const gerente = this.buscarGerentePorId(gerenteId);
+        const cliente = gerente.clientes.find(cliente => cliente.id === clienteId);
+        if (!cliente) throw new Error('Cliente não encontrado');

```

Comment:
O seguinte trecho de código:

```ts
const cliente = gerente.clientes.find(cliente => cliente.id === clienteId);
if (!cliente) throw new Error('Cliente não encontrado');
```

Está duplicado em pelo menos 3 partes dessa classe. Como sugestão para implementar o princípio DRY, você pode criar uma função responsável por buscar o cliente e retorna o erro, caso o cliente não seja encontrado!

Reply:
Refatorei diva, criei um método buscarCliente que encapsula a lógica de buscar um cliente específico associado a um gerente e lançar uma exceção se o cliente não for encontrado.
Agora os métodos removerCliente, abrirConta, fecharConta e modificarTipoConta utilizam o método buscarCliente para evitar a duplicação de código ^^